cmd/build.go

```diff
@@ -527,11 +527,6 @@ func updateDeploymentConfig(config *buildCommandConfig) error {
 
 	appsodyApplication.Spec.CreateKnativeService = &config.knative
 
-	imageName := appsodyApplication.Spec.ApplicationImage
```

Comment:
Looks like this is now going to always overwrite the image in app-deploy.yaml? What if the user had manually edited the app-deploy.yaml with their own image? Maybe we don't care and this is fine, just thinking through the changes.

Reply:
I can imagine that they would run `appsody deploy --no-build`. We could read the `app-deploy.yaml` and if they haven't specified a tag check if the applicationImage is the same as the one we are about to build.

Reply:
@chilanti what are your thoughts on this? Was there a scenario where the user would want to manually change the image in `app-deploy.yaml`?

Reply:
FYI, our current docs state the following `If the `--generate only` flag is used in conjunction with the `--tag (-t)` flag, the value of the tag is preserved in the app-deploy.yaml file and used during subsequent `appsody deploy` invocations.
`. With this change, the user will need to pass the tag flag in every invocation of `appsody build` and `deploy`.